src/adr/0032-distributed-mail-queue-cleanup.md

```diff
@@ -30,8 +30,18 @@ Failing cleanup will lead to the content being eventually updated upon next `bro
 
 ## Consequences
 
-MailQueue content will eventually be dropped both in Cassandra and in ObjectStorage once Deduplicated Blob Store garbage 
+MailQueue content will eventually be dropped both in Cassandra and in ObjectStorage once Blob Store garbage 
 collection is implemented. This will both allow reclaiming storage space, reducing related costs, and respect privacy 
 of James users.
 
 Updating browse start will then be two times more expensive.
+
+## Alternative
+
+A [proposal](https://github.com/linagora/james-project/pull/3291#pullrequestreview-393501339) was made to piggy back 
```

Comment:
I never proposed that we delete anything from `enqueuedMailsV3`. BTW, it's useless to delete entries from it as we can just drop the old slices.

Reply:
Ok, so my suggestion was: on dequeue and delete, we delete the blob from the blobstore.
To cleanup the `enqueuedMailsV3` and `deletedMailsV2` tables, we can drop entire slices which are behind `browseStart`.
We may let some dangling blobs in blobstore in case of server failure but given the frequency of the event, it may not be a problem to have some data stuck forever.
Would it work?